subprojects/core-api/src/main/java/org/gradle/api/file/SettingsLayout.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.file;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.initialization.Settings;
+
+/**
+ * Provides access to important locations for a {@link Settings} instance.
+ *
+ * <p>An instance of this type can be injected into a plugin or other object by annotating a public constructor or method with {@code javax.inject.Inject}. It is also available via {@link Settings#getLayout()}.
+ *
+ * @since 8.5
+ */
+@Incubating
+public interface SettingsLayout extends FileSystemLayout {
```

Comment:
💅 `@ServiceScope` annotation is missing

Reply:
Thanks! I am adding it (using `@ServiceScope(Scopes.Gradle.class)`), but I am concerned no tests failed. What should be the effect at runtime? I tried to figure out the magic that makes ServiceScope work, to no avail.

Reply:
The only effect is that CC will not serialize the instance of the service if it sees one. It will instead fetch the fresh version from the context on load.

But we should annotate all services with `@ServiceScope`, especially the new ones.

Reply:
Right! 

It might be good to have a test that asserts that a reference to `SettingsLayout` can be stored / loaded correctly (e.g. `storedSettingsLayout === injectedSettingsLayout`.

Reply:
Just for my understanding (I want to provide the correct value), but what is the use for the `scope` field today? I found code that checks for the presence of the annotation, but could not find code that queries the `scope` field.

Reply:
Today, the only code that relies on the `@ServiceScope` annotation is CC's serialization process. It only checks the presence of the annotation, but not the value of the scope.

My take would be that we should expose a service at the highest possible scope where it is valid to avoid introducing artificial constraints on users. Though, it seems from the user perspective the `Build` and the `Gradle` scopes are effectively equivalent, because the settings file already sees the `Gradle` scope. And AFAIK, there is no point in time where user code only observes the `Build` scope.

Also, Adam at some point emphasized that these annotations serve as an important part of the documentation. Documentation for the users -- they can understand in what scope it is valid to query the service. Documentation for Gradle engineers -- this helps with refactorings (according to Adam).

My additional take would be that all services must explicitly define the scope. Currently, only a handful of them do. And we also must validate (at least at Gradle project test-time), that the services are registered at the right scope. Right now there is no such validation, so [mismatches happen](https://github.com/gradle/gradle/pull/26082) and lead to bugs.

Reply:
Thanks, I was guessing it was meant for documentation only. The javadoc for the annotation seems to imply actual runtime semantics, which led me to believe we had a bug somewhere and the scope should have been enforced somehow (or the wording weakened).

```
 /**
 * Attached to a service interface to indicate in which scope it is defined in.
 * Services are lifecycled with their scope, and stopped/closed when the scope is closed.
 *
 * Services are visible to other services in the same scope and descendent scopes.
 * Services are not visible to services in ancestor scopes.
 *
 * @see org.gradle.internal.service.scopes.Scopes
 */
 ```

Reply:
I am leaving this as is atm as I am not convinced which scope is the right one between Gradle and Build. I am unclear also on the relationship between service registries and scopes. For instance, should services registered under a service registry implementation named FooScopeServices (such as GradleScopeServices, BuildScopeServices...) be assigned the Foo scope? What about services registered by SettingsScopeServices? (there is no Settings scope per se).